components/gateway/internal/auditlog/service.go

```diff
@@ -0,0 +1,191 @@
+package auditlog
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/gateway/internal/auditlog/model"
+	"github.com/kyma-incubator/compass/components/gateway/pkg/proxy"
+	"github.com/pkg/errors"
+)
+
+type AuditlogMessage struct {
+	Request  string
+	Response string
+	proxy.Claims
+}
+
+type AuditlogSource struct {
+	logsChannel chan AuditlogMessage
+}
+
+func NewAuditlogSink(logsChannel chan AuditlogMessage) *AuditlogSource {
+	return &AuditlogSource{
+		logsChannel: logsChannel,
+	}
+}
+
+func (sink *AuditlogSource) Log(request, response string, claims proxy.Claims) error {
+	msg := AuditlogMessage{
```

Comment:
Why do you expect to return an error from this func?

Reply:
I want to be compatible with `AuditlogService`